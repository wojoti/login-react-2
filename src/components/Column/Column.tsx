import React, { ReactNode } from "react";
import { Wrapper } from "./Column.style";

type Props = {
  children: ReactNode;
};

const Column = (props: Props) => {
  return <Wrapper>{props.children}</Wrapper>;
};

export default Column;
