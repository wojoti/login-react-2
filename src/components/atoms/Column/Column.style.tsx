import styled from 'styled-components'
interface Props {
  mt?: number
  mb?: number
  ml?: number
  mr?: number
  pt?: number
  pb?: number
  pl?: number
  pr?: number
  justify?: string
  items?: string
}

export const Wrapper = styled.div<Props>`
  margin-top: ${(props) => props.mt || 0}px;
  margin-bottom: ${(props) => props.mb || 0}px;
  margin-left: ${(props) => props.ml || 0}px;
  margin-right: ${(props) => props.mr || 0}px;
  padding-top: ${(props) => props.pt || 0}px;
  padding-bottom: ${(props) => props.pb || 0}px;
  padding-left: ${(props) => props.pl || 0}px;
  padding-right: ${(props) => props.pr || 0}px;
  display: flex;
  flex-direction: column;
  justify-content: ${(props) => props.justify || 'start'};
  align-items: ${(props) => props.items || 'start'};
`
