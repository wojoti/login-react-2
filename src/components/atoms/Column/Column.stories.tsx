import { ComponentMeta, ComponentStory } from "@storybook/react";

import Column from "./Column";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Atoms/Column",
  component: Column,
  decorators: [
    (Story) => (
      <div style={{ border: "1px solid lightgray" }}>
        <Story />
      </div>
    ),
  ],
} as ComponentMeta<typeof Column>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Column> = (args) => (
  <Column {...args}>
    <span>Example span.</span>
    <span>Example span.</span>
    <span>Example span.</span>
  </Column>
);

export const Default = Template.bind({});
Default.args = {};
