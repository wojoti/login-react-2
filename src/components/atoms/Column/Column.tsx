import React, { ReactNode } from "react";
import { Wrapper } from "./Column.style";

type Props = {
  children?: ReactNode;
  mt?: number;
  mb?: number;
  ml?: number;
  mr?: number;
  pt?: number;
  pb?: number;
  pl?: number;
  pr?: number;
  justify?: string;
  items?: string;
};

const Column = (props: Props) => {
  return (
    <Wrapper
      mt={props.mt}
      mb={props.mb}
      ml={props.ml}
      mr={props.mr}
      pt={props.pt}
      pb={props.pb}
      pl={props.pl}
      pr={props.pr}
      justify={props.justify}
      items={props.items}
    >
      {props.children}
    </Wrapper>
  );
};

export default Column;
