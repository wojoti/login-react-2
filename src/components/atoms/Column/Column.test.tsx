import { cleanup, render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Column, { ColumnProps } from "./Column";

afterEach(() => {
  cleanup();
});

const props: ColumnProps = {
  testId: "test-column-id",
};

const modifiedProps: ColumnProps = {
  ...props,
  mt: 14,
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(<Column {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render column", () => {
  render(<Column {...props} />);
  const columnElement = screen.getByTestId("test-column-id");
  expect(columnElement).toBeInTheDocument();
});

test("should render column with modified style through props", () => {
  render(<Column {...modifiedProps} />);
  const columnElement = screen.getByTestId("test-column-id");
  expect(columnElement).toBeInTheDocument();
  expect(columnElement).toHaveStyle("margin-top: 14px");
});
