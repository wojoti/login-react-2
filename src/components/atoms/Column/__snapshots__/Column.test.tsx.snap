// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin-top: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<div
  className="c0"
  data-testid="test-column-id"
/>
`;
